Sources/SwiftFormatRules/OrderedImports.swift

```diff
@@ -24,241 +27,365 @@ import SwiftSyntax
+            currentLine = Line()
+            afterNewline = true  // Note: trailing line comments always come before any newlines.
+          }
+        default:
+          if afterNewline || isFirstBlock {
+            currentLine.leadingTrivia.append(piece)  // This will be a standalone comment.
+          } else {
+            currentLine.trailingTrivia.append(piece)  // This will be a trailing line comment.
+          }
+        }
       }
     }
-    return Trivia(pieces: pieces)
+    currentLine.codeBlock = block  // This represents actual code: imports and otherwise.
+    isFirstBlock = false
   }
+  lines.append(currentLine)
 
-  enum ImportType: Int {
-    case regularImport = 0
-    case individualImport
-    case testableImport
-  }
+  return lines
+}
+
+/// This function transforms a list of Line objects into a list of CodeBlockItemSyntax objects,
+/// replacing the trivia appropriately to ensure comments appear in the right location.
+func convertToCodeBlockItems(lines: [Line]) -> [CodeBlockItemSyntax] {
+  var output: [CodeBlockItemSyntax] = []
+  var triviaBuffer: [TriviaPiece] = []
 
-  /// Indicates to which import type those the given import belongs.
-  func classifyImport(_ impStatement: CodeBlockItemSyntax) -> ImportType? {
-    guard let importToken = impStatement.firstToken else {return nil}
-    guard let nextToken = importToken.nextToken else {return nil}
-  
-    if importToken.tokenKind == .atSign && nextToken.text == "testable" {
-      return ImportType.testableImport
+  for line in lines {
+    triviaBuffer += line.leadingTrivia
+    if let block = line.codeBlock {
+      // Comments and newlines are always located in the leading trivia of an AST node, so we need
+      // not deal with trailing trivia.
+      output.append(
+        replaceTrivia(
+          on: block,
+          token: block.firstToken,
+          leadingTrivia: Trivia(pieces: triviaBuffer)
+        ) as! CodeBlockItemSyntax
+      )
+      triviaBuffer = []
+      triviaBuffer += line.trailingTrivia
     }
-    if nextToken.tokenKind != .identifier(nextToken.text) {
-      return ImportType.individualImport
+
+    // Merge multiple newlines together into a single trivia piece by updating it's N value.
+    if let lastPiece = triviaBuffer.last, case .newlines(let N) = lastPiece {
+      triviaBuffer[triviaBuffer.endIndex - 1] = TriviaPiece.newlines(N + 1)
+    } else {
+      triviaBuffer.append(TriviaPiece.newlines(1))
     }
-    return ImportType.regularImport
   }
 
-  /// Return the name of the given import group type.
-  func importTypeName(_ type: Int) -> String {
-    switch type {
-    case 0:
-      return "regular imports"
-    case 1:
-      return "Individual declaration imports"
-    default:
-      return "@testable imports"
-    }
+  return output
+}
+
+enum LineType {
+  case `import`
+  case indivimport
+  case testimport
+  case code
+  case comment
+  case blankline
+}
+
+func lineTypeName(_ type: LineType) -> String {
```

Comment:
You can make `LineType` conform to `CustomStringConvertible` and then implement the `description` property to return these strings instead; then you can just write `"Place \(before) imports before \(after) imports"` in the diagnostic string below.

Can you also make these strings all lowercase, since they're inserted into the middle of a sentence?

Reply:
Done.